README.rst

```diff
@@ -686,13 +686,17 @@ variable ``FUN``:
         is given, then all the ``bin`` passed will be prepended with this
         ``path``. ``bin``\ s can be passed as multiple arguments or as one
         list.
-``mapfiles(src, dst, file[, ...][, append_suffix=True])``
+``mapfiles(src, dst, [file[, ...]][, append_suffix=True])``
         A very simple function that just returns a list with
         ``{src}/{file}={dst}/{file}{VAR.suffix}`` for each ``file`` passed.
-        ``file``\ s can be passed as multiple arguments or as one list. A named
-        argument ``append_suffix`` can be passed at the end to control whether
-        ``VAR.suffix`` is appended to each destination file. ``append_suffix``
-        defaults to ``True`` if not given.
+        ``file``\ s can be passed as multiple arguments or as one list. If no
+        ``file``\ s are passed, a simple ``{src}/={dst}/`` mapping is returned
+        which results in the directory structure under ``src`` (including all
+        contained files) being replicated as-is from the local filesystem into
+        ``dst`` within the package. A named argument ``append_suffix`` can be
+        passed at the end to control whether ``VAR.suffix`` is appended to each
+        destination file. ``append_suffix`` defaults to ``True`` if not given,
+        and is only applicable when at least one ``file`` is passed.
```

Comment:
>And honestly using FUN.mapfiles('src', 'dst') vs using "src/=dst/" makes little to no sense (or am I missing something here?)

I created a package definition file which initially had something like:
```
ARGS = FUN.mapfiles(VAR.bindir, '/usr/bin', app1, app2)
ARGS += FUN.mapfiles('configs', '/etc/configs', config1, config2) + [
    'helpers/=/etc/helpers'
]
```

In an attempt to make it more "symmetrical", I thought I could instead use:
```
ARGS = FUN.mapfiles(VAR.bindir, '/usr/bin', app1, app2)
ARGS += FUN.mapfiles('configs', '/etc/configs', config1, config2) 
ARGS += FUN.mapfiles('helpers', '/etc/helpers')
```

I figured that at best it would do the most intuitive thing - exactly replicate everything under the local `helpers` directory in the remote `/etc/helpers` directory; and at worst give me a build error on the third line saying something like "insufficient number of arguments given". But neither of these happened, and it returned an empty list instead. This, imo, is a bug.

I thought the best way to fix the bug is to provide what I consider the most intuitive behaviour when no file names are explicitly passed. So although technically there's no difference between using `FUN.mapfiles('src', 'dst')` and `src/=dst/`, I feel that the former way improves readability by providing a consistent look to the code.